dm/ctl/master/operate_task.go

```diff
@@ -0,0 +1,184 @@
+	}
+
+	name := common.GetTaskNameFromArgOrFile(cmd.Flags().Arg(0))
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return err
+	}
+
+	resp, err := common.OperateTask(taskOp, name, sources)
+	if err != nil {
+		common.PrintLinesf("can not %s task %s", strings.ToLower(taskOp.String()), name)
+		return err
+	}
+
+	common.PrettyPrintResponse(resp)
+	return nil
+}
+
+func addOperateSourceTaskFlags(cmd *cobra.Command) {
+	// control workload to dm-cluster for sources with large number of tasks.
+	cmd.Flags().Int(batchSizeFlag, defaultBatchSize, "batch size when operating all (sub)tasks bound to a source")
+}
+
+func operateSourceTaskFunc(taskOp pb.TaskOp, cmd *cobra.Command) error {
+	source, batchSize, ok := parseOperateSourceTaskParams(cmd)
+	if !ok {
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	sources := []string{source}
+	ctx, cancel := context.WithTimeout(context.Background(), common.GlobalConfig().RPCTimeout)
+	defer cancel()
+
+	req := pb.QueryStatusListRequest{Sources: sources}
+	resp := &pb.QueryStatusListResponse{}
+	if err := common.SendRequest(ctx, "QueryStatus", &req, &resp); err != nil {
+		common.PrintLinesf("cannot query status of source: %v", sources)
+		return err
+	}
+
+	if !resp.Result || len(resp.Sources) == 0 {
+		common.PrettyPrintInterface(&batchTaskResult{Result: false, Msg: resp.Msg, Tasks: []*operateTaskResult{}})
+		return nil
+	}
+
+	result := batchOperateTask(taskOp, batchSize, sources, resp.Sources[0].SubTaskStatus)
+	common.PrettyPrintInterface(result)
+
+	return nil
+}
+
+func batchOperateTask(taskOp pb.TaskOp, batchSize int, sources []string, subTaskStatus []*pb.SubTaskStatus) *batchTaskResult {
+	result := batchTaskResult{Result: true, Tasks: []*operateTaskResult{}}
+
+	if len(subTaskStatus) < batchSize {
+		batchSize = len(subTaskStatus)
+	}
+
+	workCh := make(chan string)
+	go func() {
+		for _, subTask := range subTaskStatus {
+			workCh <- subTask.Name
+		}
+		close(workCh)
+	}()
+
+	var wg sync.WaitGroup
+	resultCh := make(chan *operateTaskResult, 1)
+	for i := 0; i < batchSize; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for name := range workCh {
+				taskResult := operateTaskResult{Task: name, Op: taskOp.String()}
+				taskOpResp, err := common.OperateTask(taskOp, name, sources)
+				if err != nil {
+					taskResult.Result = false
+					taskResult.Msg = err.Error()
+				} else {
+					taskResult.Result = taskOpResp.Result
+					taskResult.Msg = taskOpResp.Msg
+					taskResult.Sources = taskOpResp.Sources
+				}
+				resultCh <- &taskResult
+			}
+		}()
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultCh)
+	}()
+
+	for item := range resultCh {
+		result.Tasks = append(result.Tasks, item)
+	}
+
+	sort.Slice(result.Tasks, func(i, j int) bool {
+		return result.Tasks[i].Task < result.Tasks[j].Task
+	})
+
+	return &result
+}
+
+func parseOperateSourceTaskParams(cmd *cobra.Command) (string, int, bool) {
```

Comment:
I think is better to return a error instead of a bool so the caller can further wrap the error detail if needed